extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -363,8 +362,6 @@ export class AmpA4A extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    const adType = this.element.getAttribute('type');
-    this.config = adConfig[adType] || {};
```

Comment:
If we're expecting that objects have this.config variable then I suggest we make it explicitly part of the constructor of AmpA4A.  I'm not a fan of this expectation that the object just exist in implementations.  Seems unclear to me and future implementors will likely not be aware of how to integrate.

I would instead propose a new, explicit, overrideable function:

/** @return {!Array<string>} */
getPreconnectUrls() { return []; }
/** @return {!Array<string>} */
getPrefetchUrls() { return []; }